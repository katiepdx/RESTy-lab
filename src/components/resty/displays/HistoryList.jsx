import React from 'react'
import propTypes from 'prop-types'
import HistoryItem from './HistoryItem'

const HistoryList = ({ history }) => {
  return history.map(search => (
    <>
      <HistoryItem key={search.url + search.method} {...search} />
    </>
  ))
}

HistoryList.propTypes = {
  // specify shape of data
  history: propTypes.arrayOf(propTypes.shape({
    url: propTypes.string.isRequired,
    method: propTypes.string.isRequired
  }))
}

export default HistoryList
