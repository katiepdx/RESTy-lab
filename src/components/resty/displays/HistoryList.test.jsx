import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList'

describe('HistoryList component', () => {
  afterAll(() => cleanup());

  it('renders HistoryList', () => {
    const { asFragment } = render(<HistoryList
      history={[
        {
          url: "www.api1.com",
          method: "GET"
        }, {
          url: "www.api2.com",
          method: "PUT"
        }, {
          url: "www.api3.com",
          method: "GET"
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  })
});
