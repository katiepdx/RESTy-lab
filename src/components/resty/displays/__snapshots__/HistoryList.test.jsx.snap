// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders HistoryList 1`] = `
<DocumentFragment>
  <section
    class="historyItem"
  >
    <p>
      Method: GET
    </p>
    <p>
      URL: www.api1.com
    </p>
  </section>
  <section
    class="historyItem"
  >
    <p>
      Method: PUT
    </p>
    <p>
      URL: www.api2.com
    </p>
  </section>
  <section
    class="historyItem"
  >
    <p>
      Method: GET
    </p>
    <p>
      URL: www.api3.com
    </p>
  </section>
</DocumentFragment>
`;
